packages/logging_cloudwatch/amplify_logging_cloudwatch/lib/src/queued_item_store/index_db/indexed_db_adapter.dart

```diff
@@ -0,0 +1,177 @@
+      };
+    _database = await openRequest.future;
+  }
+
+  /// Returns a new [IDBObjectStore] instance after waiting for initialization
+  /// to complete.
+  IDBObjectStore _getObjectStore() {
+    final transaction = _database.transaction(
+      storeName,
+      mode: IDBTransactionMode.readwrite,
+    );
+    final store = transaction.objectStore(storeName);
+    return store;
+  }
+
+  @override
+  Future<void> addItem(String string, String timestamp) async {
+    _currentTotalByteSize += utf8.encode(string).length;
+    _currentTotalByteSize += utf8.encode(timestamp).length;
+    _currentTotalByteSize += 8; // 8 bytes for the id
+    await _databaseOpenEvent;
+    await _getObjectStore().add(string, timestamp).future;
+  }
+
+  @override
+  Future<Iterable<QueuedItem>> getCount([int? count]) async {
+    final readValues = <QueuedItem>[];
+
+    await _databaseOpenEvent;
+    final store = _getObjectStore();
+    final request = store.getAll(null, count);
+
+    await request.future;
+
+    for (final elem in request.result) {
+      final value = elem as Object;
+      final id = getProperty<int>(value, 'id');
+      final string = getProperty<String>(value, 'value');
+      final timestamp = getProperty<String>(value, 'timestamp');
+      readValues.add(
+        QueuedItem(id: id, value: string, timestamp: timestamp),
+      );
+    }
+    return readValues;
+  }
+
+  @override
+  Future<void> deleteItems(Iterable<QueuedItem> items) async {
+    if (items.isEmpty) return;
+
+    _currentTotalByteSize -= utf8.encode(items.first.value).length;
+    _currentTotalByteSize -= utf8.encode(items.first.timestamp).length;
+    _currentTotalByteSize -= 8; // 8 bytes for the id
+
+    final idsToDelete = items.map((item) => item.id);
+
+    await _databaseOpenEvent;
+    final store = _getObjectStore();
+
+    final ranges = idsToDelete.splitBetween((a, b) => b != a + 1).map(
+          (range) => IDBKeyRange.bound(range.first, range.last),
+        );
+    await Future.wait<void>(
+      ranges.map((range) => store.deleteByKeyRange(range).future),
+    );
+  }
+
+  @override
+  Future<Iterable<QueuedItem>> getAll() async {
+    final readValues = <QueuedItem>[];
+
+    await _databaseOpenEvent;
+    final store = _getObjectStore();
+    final request = store.getAll(null, null);
+
+    await request.future;
+
+    for (final elem in request.result) {
+      final value = elem as Map<String, dynamic>;
+      final id = value['id'] as int;
+      final itemValue = value['value'] as String;
+      final timestamp = value['timestamp'] as String;
+      readValues.add(
+        QueuedItem(id: id, value: itemValue, timestamp: timestamp),
+      );
+    }
+
+    return readValues;
+  }
+
+  @override
+  Future<bool> isFull(int maxSizeInMB) async {
+    final maxBytes = maxSizeInMB * 1024 * 1024;
+    return _currentTotalByteSize >= maxBytes;
+  }
+
+  /// Clear the database.
+  @override
+  Future<void> clear() async {
+    _currentTotalByteSize = 0;
```

Comment:
This should be the last line, as it's only true if the two futures below succeed